### ISSUE TITLE
missing definitions, constructors, and so much more

### ISSUE THREAD

**Issue by AbhimanyuAryan** [dotnet/core#481]
I am a Ruby(& RoR) developer. So after dotnet core went cross platform, I keep on continuously learning CSharp from MVA but I find it hard to follow MVA's tutorials. 

Reason: I am a mac/linux(not bash of Windows) user. I don't like Windows(bit harsh to say this in MS community but true)

I know dotnet core is subset of .net framework but there's gotta be some way to use these definitions & stuff in .dotnet core 

Say I don't have StreamReader('file.json') constructor in dotnet core. 

Why so? What should I do to learn C# Sharp? Stop following MVA tutorials & read only C# Sharp(dotnet core) documentation or there's a possible work around for things like this?

Or I can import .net stuff(libraries/methods/classes) to dotnet core?

**Comment by Petermarcu** [comment]
@AbhimanyuAryan , good feeedback and we're really glad you are interested in and trying out C#. 

There are a few things happening. As with any project, it had to start somewhere and .NET Core started "small". Since 1.0, we have heard strong feedback that many of the types people know an love need to be available. So in .NET Core 2.0 we're adding a ton of them back.

There is a site here that will show you where API's are available. I added a link to the example you gave. We're adding it in .NET Core 2.0. https://apisof.net/catalog/System.IO.StreamReader..ctor(String)

Early builds of .NET Core 2.0 and its SDK are available here: https://github.com/dotnet/cli/tree/master

There is also a big effort called .NET Standard which will start to ensure you can reason about which API's are available everywhere and force more consistency. 

One last thing is that we're looking at ways to just add references to libraries built for .NET Framework from .NET Core apps. They may or may not work depending on if they call into Windows API's or API's that just can't be brought to .NET Core. But they may also work and we want people to be able to try and unblock themselves.

Hopefully this helps. Again. Good feedback and I hope we can continue to make this easier.
